backend/tenantfirstaid/session.py

```diff
@@ -29,44 +42,56 @@ def __init__(self):
         except Exception as e:
             print(e)
 
-    def get(self):
+    # Retrieves the session ID from Flask session or creates a new one
+    def get_flask_session_id(self) -> str:
         session_id = session.get("session_id")
         if not session_id:
-            return self.getNewSessionData()
+            session_id = str(uuid.uuid4())
+            session["session_id"] = session_id
+
+            @after_this_request
+            def save_session(response):
+                session.modified = True
+                return response
+
+        return session_id
+
+    def get(self) -> TenantSessionData:
+        session_id = self.get_flask_session_id()
 
         saved_session = self.db_con.get(session_id)
```

Comment:
FYI, there's a known issue (tl;dr - union of sync & async return types is nonsensical) about `valkey` typing ... https://github.com/valkey-io/valkey-py/issues/164